tests/integration/relationships/has-many-test.js

```diff
@@ -99,6 +100,80 @@ module("integration/relationships/has_many - Has-Many Relationships", {
   }
 });
 
+test("When an object is destroyed, it's hasMany parent is updated", function(assert) {
+  assert.expect(3);
+  let done = assert.async();
+
+  var Post = DS.Model.extend({
+    comments: DS.hasMany('comment', { async: true }),
+
+    hasComments: computed('comments.[]', function() {
+      return this.hasMany('comments').ids().length > 0;
+    })
+  });
+  Post.reopenClass({ toString: () => 'Post' });
+
+  var Comment = DS.Model.extend({
+    post: DS.belongsTo('post', { async: true })
+  });
+  Comment.reopenClass({ toString: () => 'Comment' });
+
+  env = setupStore({
+    post: Post,
+    comment: Comment,
+    adapter: DS.RESTAdapter.extend({
+      shouldBackgroundReloadRecord: () => false
+    })
+  });
+
+  var commentId = 1;
+  env.registry.register('adapter:comment', DS.RESTAdapter.extend({
+    deleteRecord(record) {
+      return Ember.RSVP.resolve();
+    },
+    updateRecord(record) {
+      return Ember.RSVP.resolve();
+    },
+    createRecord() {
+      return Ember.RSVP.resolve({ comments: { id: commentId++ }});
+    }
+  }));
+
+  run(function() {
+    env.store.push({
+      data: [{
+        type: 'post',
+        id: '1',
+        relationships: {
+          comments: {
+            data: [
+              { type: 'comment', id: '1' }
+            ]
+          }
+        }
+      }, {
+        type: 'comment',
+        id: '1'
+      }]
+    });
+  });
+
+  run(function() {
+    env.store.findRecord('post', 1).then(function (post) {
+      assert.ok(post.get('hasComments'));
+
+      env.store.findRecord('comment', 1).then(function (comment) {
+        assert.ok(comment);
+
+        comment.destroyRecord().then(function() {
+          assert.notOk(post.get('hasComments'));
+          done();
```

Comment:
Updated - actually returning `run` caused issues but the way it is now seems to be working (properly failing)